app/proguard-rules.pro

```diff
@@ -0,0 +1,155 @@
+-dontwarn org.bouncycastle.mail.smime.handlers.x_pkcs7_mime
+-dontwarn org.bouncycastle.mail.smime.handlers.x_pkcs7_signature
+-dontwarn org.bouncycastle.mail.smime.SMIMECompressed
+-dontwarn org.bouncycastle.mail.smime.SMIMECompressedGenerator
+-dontwarn org.bouncycastle.mail.smime.SMIMECompressedGenerator$1
+-dontwarn org.bouncycastle.mail.smime.SMIMECompressedGenerator$ContentCompressor
+-dontwarn org.bouncycastle.mail.smime.SMIMECompressedParser
+-dontwarn org.bouncycastle.mail.smime.SMIMEEnveloped
+-dontwarn org.bouncycastle.mail.smime.SMIMEEnvelopedGenerator
+-dontwarn org.bouncycastle.mail.smime.SMIMEEnvelopedGenerator$1
+-dontwarn org.bouncycastle.mail.smime.SMIMEEnvelopedGenerator$ContentEncryptor
+-dontwarn org.bouncycastle.mail.smime.SMIMEEnvelopedParser
+-dontwarn org.bouncycastle.mail.smime.SMIMEGenerator
+-dontwarn org.bouncycastle.mail.smime.SMIMESigned
+-dontwarn org.bouncycastle.mail.smime.SMIMESigned$1
+-dontwarn org.bouncycastle.mail.smime.SMIMESignedGenerator
+-dontwarn org.bouncycastle.mail.smime.SMIMESignedGenerator$1
+-dontwarn org.bouncycastle.mail.smime.SMIMESignedGenerator$ContentSigner
+-dontwarn org.bouncycastle.mail.smime.SMIMESignedParser
+-dontwarn org.bouncycastle.mail.smime.SMIMESignedParser$1
+-dontwarn org.bouncycastle.mail.smime.SMIMEToolkit
+-dontwarn org.bouncycastle.mail.smime.SMIMEUtil
+-dontwarn org.bouncycastle.mail.smime.SMIMEUtil$LineOutputStream
+-dontwarn org.bouncycastle.mail.smime.SMIMEUtil$WriteOnceFileBackedMimeBodyPart
+-dontwarn org.bouncycastle.mail.smime.util.FileBackedMimeBodyPart
+-dontwarn org.bouncycastle.mail.smime.util.SharedFileInputStream
+-dontwarn org.bouncycastle.mail.smime.validator.SignedMailValidator
+-dontwarn org.bouncycastle.x509.util.LDAPStoreHelper
+-dontwarn org.h2.fulltext.FullTextLucene
+-dontwarn org.h2.fulltext.FullTextLucene$FullTextTrigger
+-dontwarn org.h2.fulltext.FullTextLucene$IndexAccess
+-dontwarn org.h2.index.SpatialTreeIndex
+-dontwarn org.h2.jdbcx.JdbcDataSource
+-dontwarn org.h2.jdbcx.JdbcDataSourceFactory
+-dontwarn org.h2.jdbcx.JdbcXAConnection
+-dontwarn org.h2.jdbcx.JdbcXid
+-dontwarn org.h2.jmx.DatabaseInfo
+-dontwarn org.h2.jmx.DocumentedMBean
+-dontwarn org.h2.mvstore.db.MVSpatialIndex
+-dontwarn org.h2.server.web.DbStarter
+-dontwarn org.h2.server.web.WebServlet
+-dontwarn org.h2.tools.Console
+-dontwarn org.h2.util.AbbaLockingDetector
+-dontwarn org.h2.util.AbbaLockingDetector$1
+-dontwarn org.h2.util.DbDriverActivator
+-dontwarn org.h2.util.JdbcUtils
+-dontwarn org.h2.util.OsgiDataSourceFactory
+-dontwarn org.h2.util.Profiler
+-dontwarn org.h2.util.SourceCompiler
+-dontwarn org.h2.util.SourceCompiler$ClassFileManager
+-dontwarn org.h2.util.SourceCompiler$ClassFileManager$1
+-dontwarn org.h2.util.SourceCompiler$JavaClassObject
+-dontwarn org.h2.util.SourceCompiler$StringJavaFileObject
+-dontwarn org.h2.util.ThreadDeadlockDetector
+-dontwarn org.h2.util.Utils
+-dontwarn org.h2.value.ValueGeometry
+-dontwarn org.h2.value.ValueGeometry$ZVisitor
```

Comment:
Pretty sure you can do `-dontwarn org.h2.*` or something like that (e.g. wildcard for the whole package).

Reply:
Yes, that's possible, but I prefer to document the warnings using that. I planned to replace the database by sqlite (which is bundled with Android and the default database engine in many apps) which allows to remove many dontwarn entries.

Reply:
Yes that definitely makes sense I think.

Reply:
I'd prefer to keep it simple, and having that many lines isn't really simple. But if you prefer, we can go with this.

And yes, using the native Android database is probably a good idea.